server/src/main/scala/de/frosner/broccoli/websocket/WebSocketMessageHandler.scala

```diff
@@ -58,3 +63,53 @@ class BroccoliMessageHandler @Inject()(instances: NomadInstances, instanceServic
           .fold(GetInstanceTasksError(instanceId, _), GetInstanceTasksSuccess)
     }
 }
+
+/**
+  * Cache some messages from the Broccoli message handler.
+  *
+  * Reduces load on Nomad.
+  *
+  * @param underlying The underlying message handler.
+  */
+class CachedBroccoliMessageHandler @Inject()(
+    underlying: BroccoliMessageHandler,
+    cache: CacheApi
+)(implicit ec: ExecutionContext)
+    extends WebSocketMessageHandler {
+
+  private val timeout: Duration = 3.seconds
```

Comment:
This is how I understand the mechanism: Instead of limiting the clients in the amount of requests they can make to Broccoli but then forward every request to nomad directly we are allowing the clients to make as many requests as they like, but then cache the nomad response for three seconds. Is this correct?

I think we should make the cache timeout configurable so we can find an optimal tradeoff between the up-to-dateness and the load we put on Nomad. What do you think?

Reply:
That's the approach, yes.  I'll add a configuration value for the timeout.